### ISSUE TITLE
Use syntax table? rather than manual search span with text { }

### ISSUE THREAD

**Issue by cheunghy** [jrattray13/swackets#5]
IMHO, manual search span for { and } is not efficient, and it's display code rather than model code.

I'm a new user to atom. I think there maybe some core model like syntax table and overlay. It may provide better performance and extensibility for example ruby use `def` and `end` rather than `{` and `}` when defining a method.

**Comment by jrattray13** [comment]
Yeah I know this is not ideal, currently swackets is just an experimental proof of concept.

There are two approaches which potentially will be better than manual search,

1) Decorations atom API currently this works well but does not allow you to add text-color to the marker overlay, only background color and underlining. I am looking into having decorations as a toggle-option feature in next release but will not have it as default until they allow original purpose of swackets (color coded brackets) since Atom text editor is only version 1, perhaps this may be added  in later releases.

2) Syntax (Or grammar) however I have huge doubt about the infeasibility of this, because it's regex. And i'm not sure regex is powerful enough to handle complicated cases of nested brackets. 

I am looking into both of these but for now just treat swackets as experimental, once I add decorations perhaps it will be more a polished package :smile:  Thanks for suggestion.

**Comment by cheunghy** [comment]
Thanks!
